### PR TITLE
refactor: [M3-6903] - Replace Select with Autocomplete in: volumes

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝
This PR eliminates redundant logic and reuses ConfigSelect in AttachVolumeDrawer. 
Replaced `Select` with Autocomplete in `ConfigSelect` component.
Reused `ConfigSelect` instead of Select in `AttachVolumeDrawer` component. 

## Target release date 🗓️
6/10

### Verification steps
(How to verify changes)
- Navigate to http://localhost:3000/volumes and click on attach menu option.
- Verify there is no regression in volume attach / detaching behaviors. 
- Verify there is no regression in Volume create flow.
- Navigate to http://localhost:3000/linodes/< Linode_id>/storage - Verify there is no regression in  creating / attaching / detaching volumes. 

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by jdamore-linode — APPROVED** [review]
Looks good! Confirmed that Volume create/attach/detach flows are working as expected, and messed around with the autocomplete for a while and didn't see anything out of the ordinary. Thanks @cpathipa!

File: packages/manager/src/features/Volumes/VolumeDrawer/ConfigSelect.tsx
Comment (bnussman-akamai) [inline packages/manager/src/features/Volumes/VolumeDrawer/ConfigSelect.tsx:1]: Now that we use an `Autocomplete`, we don't need to keep the `options` in the shape of `{ label: string, value: number }`.  Can we update remove `configList` and pass `configs` directly to the `Autocomplete`?

**Review by dwiley-akamai — COMMENTED** [review]
Creating, attaching, and detaching function as expected ✅

If you open the Attach drawer but don't select a linode for a few seconds, this error pops up:

Screenshot 2024-05-08 at 5 03 42 PM

There's also more space between the fields on this branch compared to prod.

**Comment by hkhalil-akamai** [comment]
> If you open the Attach drawer but don't select a linode for a few seconds, this error pops up:

I'm also observing this. Also, the spacing with the error appears off, there seems to be some extra padding on the left side.

File: packages/manager/src/features/Volumes/AttachVolumeDrawer.tsx [hunk 1]
```diff
@@ -58,27 +58,10 @@ export const AttachVolumeDrawer = React.memo((props: Props) => {
       });
     },
     validateOnBlur: false,
-    validateOnChange: false,
```
[the remark this answers is not in the document]
Comment (cpathipa) [reply packages/manager/src/features/Volumes/AttachVolumeDrawer.tsx:61]: @hkhalil-akamai Good catch, addressed this issue in the commit [b8dbb10](https://github.com/linode/manager/pull/10437/commits/b8dbb10c003180f510d1a9cc21dc7e4d9bafe2db)

**Comment by cpathipa** [comment]
> Creating, attaching, and detaching function as expected ✅
> 
> If you open the Attach drawer but don't select a linode for a few seconds, this error pops up:
> 
> Screenshot 2024-05-08 at 5 03 42 PM
> 
> There's also more space between the fields on this branch compared to prod.

@dwiley-akamai @hana-linode This issue was fixed in the commit [b8dbb10](https://github.com/linode/manager/pull/10437/commits/b8dbb10c003180f510d1a9cc21dc7e4d9bafe2db)

**Review by dwiley-akamai — APPROVED** [review]
The issue I noted previously has been fixed ✅

Another issue I noticed (that also happens in prod) is when you try to detach a volume from the Linode Storage tab, the table data doesn't get updated accordingly without a refresh.

**Review by hkhalil-akamai — APPROVED** [review]
Seeing all regressions fixed! Thanks @cpathipa!

**Pull request merged**